### PR TITLE
Heading block: move alignment controls to toolbar (#2)

### LINKED ISSUE #16648: Unify text alignment location between Heading and Paragraph blocks
The text alignment for a Paragraph block is located right in the toolbar. When making a text alignment change to a Heading, I expected it to be located similarly in the toolbar as well. 

Can we unify the placement of the alignment options? Maybe we can move the text alignment options for the Heading block to the block's toolbar as well?

**Screenshots**

text-alignment

### PR DESCRIPTION
**Pull request by gziolo**
## Description
An alternative to #16682.

Closes #16648.

Added an alignment toolbar component to the heading block's toolbar.

Screen Shot 2019-09-12 at 13 04 45
Screen Shot 2019-09-12 at 13 05 03

On the implementation side of things, the biggest difference in comparison to #16682, this PR adds an icon for each heading level which removes the need to drill down `subscript` prop to other components together with CSS styling based on the data attribute.

## TODO

- [ ] Provide better icons which have fixed size 20x20

### CODE REVIEW

**Review by youknowriad — APPROVED** [review]
Code wise, this looks good. Design-wise, these svgs might need to be tweaked but I leave that for designers.

**Comment by gziolo** [comment]
> Now that the heading level is in a drop-down, there is an open question around the heading levels, though. So as to not keep up this PR, this is fine to look at separately, but worth mentioning. And that is: should we show all heading levels, h1 to h6, in the drop-down? Given the space is there, the absence seems confusing.
> I realize the reason they are not there is that we don't want to encourage wrongful usage of the heading levels breaking the semantics. In the past we forced users into the sidebar to choose those levels. At the same time if we show all levels here, we don't need to also have the levels in the sidebar, and we do have tools to help you understand wrongful usage of heading levels, so maybe it's okay?

I'm open to iterations here. There is one thing to consider, we want to provide tooling which helps to emphasize issues with the semantics of your content. Heading levels is one of the common issues which impacts accessibility. We have this PR proposed: #14889 which adds heading hierarchy checker:

image

So, if we were to consolidate those levels in the toolbar, we would have to find a different way to present this message.

**Comment by jasmussen** [comment]
Excellent point. That is an argument to move forward with this PR as is, and explore the changes discussed separately.

And regarding those changes, it seems like it should be possible to show such a warning directly in the drop-down, no?

**Comment by jasmussen** [comment]
GIF:

heading

The icons are a bit small

before

So I increased their size

Screenshot 2019-09-13 at 15 20 04

**Review by jasmussen — APPROVED** [review]
I think we should follow up on showing all heading levels, with warnings for picking the wrong ones, but in a separate PR. As is, this is good from a design perspective. If anyone wants to give additional sanity checks to the code feel free, but otherwise 👍 👍

**Comment by jasmussen** [comment]
Here's a very rough mockup of what it could look like, to show the warnings for incorrect heading level usage in the menu:

Screenshot 2019-09-13 at 15 26 24

☝️ We would want to either link to more information, or show more information as a tooltip, or rephrase to be more descriptive. Also the h1 icon should be used, obviously. But as an "inspector" mockup, it hopefully gets the point across. What do you think?

**Comment by mapk** [comment]
As is, it feels good to me too. I agree with @jasmussen that I'd like to see all heading levels in the popover as well. Should we include all headings in this PR right now? And work in the _warnings_ in another PR as mentioned?

**Comment by youknowriad** [comment]
> Should we include all headings in this PR right now? And work in the warnings in another PR as mentioned?

i feel we should do both these things at the same time to avoid the potential "regression" of this message not being seen by people using the new dropdown in the meantime.

**Comment by gziolo** [comment]
I'm merging this PR as is to avoid lengthy debate about implications of the change in UI. For reference, this is how it would look in the dropdown:

Screen Shot 2019-09-14 at 10 15 06

**Comment by gziolo** [comment]
Continuing on the previous topic of accessibility violations.

> Here's a very rough mockup of what it could look like, to show the warnings for incorrect heading level usage in the menu:
> 
> Screenshot 2019-09-13 at 15 26 24

The biggest challenge I see is the discovery of this message. As soon as you change the level of heading, you won't see the dropdown anymore so it'd be very likely you would miss it. I'm wondering whether you see an elegant way to extend block's UI to have a space for such warnings. The other example is the contrast warning which is located in the sidebar. In that case, the feedback loop is much better as you need to perform all interactions in the same place.

File: packages/block-library/src/heading/heading-toolbar.js [hunk 1]
```diff
@@ -10,22 +10,32 @@ import { __, sprintf } from '@wordpress/i18n';
...
+		const { isCollapsed = true, minLevel, maxLevel, selectedLevel, onChange } = this.props;
+
 		return (
-			<Toolbar controls={ range( minLevel, maxLevel ).map( ( index ) => this.createLevelControl( index, selectedLevel, onChange ) ) } />
+			<Toolbar
+				isCollapsed={ isCollapsed }
+				icon={ <HeadingLevelIcon level={ selectedLevel } /> }
+				controls={ range( minLevel, maxLevel ).map(
+					( index ) => this.createLevelControl( index, selectedLevel, onChange )
+				) } />
```
[the remark this answers is not in the document]
Comment (gziolo) [reply packages/block-library/src/heading/heading-toolbar.js]: I missed that, sorry :(

**Pull request merged**